backend/app/api/dependencies.py

```
@@ -0,0 +1,58 @@
+
+from app.core.database import get_db
+from app.core.role import Role
+from app.core.security import decode_access_token
+from app.models.user import User
+
+oauth_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/users/login")
+
+
+async def get_current_user(
+        token: str = Depends(oauth_scheme),
+        db: AsyncSession = Depends(get_db())
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)